File: src/codewhispererChat/storages/chatSession.ts
/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { ChatSession } from '../clients/chat/v0/chat'

export class ChatSessionStorage {
    private sessions: Map<string, ChatSession> = new Map()

    public getSession(tabID: string): ChatSession {
        const sessionFromStorage = this.sessions.get(tabID)
        if (sessionFromStorage !== undefined) {
            return sessionFromStorage
        }

        const newSession = new ChatSession()
        this.sessions.set(tabID, newSession)

        return newSession
    }

    public deleteSession(tabID: string) {
        this.sessions.delete(tabID)
    }
}

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/clients/chat/ChatSession.kt
// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package software.aws.toolkits.jetbrains.services.cwc.clients.chat

import kotlinx.coroutines.flow.Flow
import software.aws.toolkits.jetbrains.services.cwc.clients.chat.model.ChatRequestData
import software.aws.toolkits.jetbrains.services.cwc.clients.chat.model.ChatResponseEvent

/**
 * Interface for the API that interacts with the CodeWhispererChat service. Enables sending queries to the service
 * and receiving responses.
 */
interface ChatSession {
    val conversationId: String?
    fun chat(data: ChatRequestData): Flow<ChatResponseEvent>
}
